src/test/java/com/iexec/worker/compute/ComputeExitCauseServiceTests.java

```diff
@@ -32,62 +36,98 @@ void before() {
         computeExitCauseService = new ComputeExitCauseService();
     }
 
-    //region getPreComputeExitCauseAndPrune
+    //region setAndGetPreComputeExitCauseAndPrune
     @Test
-    void setAndGetPreComputeExitCauseAndPrune() {
-        ReplicateStatusCause cause =
-                ReplicateStatusCause.PRE_COMPUTE_DATASET_URL_MISSING;
-        Assertions.assertThat(computeExitCauseService.setExitCause(ComputeStage.PRE,
-                CHAIN_TASK_ID,
-                cause)).isTrue();
-        Assertions.assertThat(computeExitCauseService.getPreComputeExitCauseAndPrune(CHAIN_TASK_ID))
-                .isEqualTo(cause);
-        Assertions.assertThat(computeExitCauseService.getReplicateStatusCause(ComputeStage.PRE,
-                CHAIN_TASK_ID)).isNull();
-    }
+    void setAndGetPreComputeExitCauses() {
+        // Test single cause
+        ReplicateStatusCause singleCause = ReplicateStatusCause.PRE_COMPUTE_DATASET_URL_MISSING;
+        Assertions.assertThat(computeExitCauseService.setExitCausesForGivenComputeStage(ComputeStage.PRE,
+                CHAIN_TASK_ID, List.of(singleCause))).isTrue();
+        Assertions.assertThat(computeExitCauseService.getExitCausesAndPruneForGivenComputeStage(ComputeStage.PRE, CHAIN_TASK_ID))
+                .isEqualTo(List.of(singleCause));
 
-    @Test
-    void shouldReturnUnknownIssueWhenPreComputeExitCauseNotSet() {
-        ReplicateStatusCause cause = computeExitCauseService.getPreComputeExitCauseAndPrune(CHAIN_TASK_ID);
-        Assertions.assertThat(cause)
-                .isNotNull()
-                .isEqualTo(ReplicateStatusCause.PRE_COMPUTE_FAILED_UNKNOWN_ISSUE);
+        // Test multiple causes in single call
+        List<ReplicateStatusCause> multipleCauses = Arrays.asList(
+                ReplicateStatusCause.PRE_COMPUTE_DATASET_URL_MISSING,
+                ReplicateStatusCause.PRE_COMPUTE_INVALID_DATASET_CHECKSUM
+        );
+        Assertions.assertThat(computeExitCauseService.setExitCausesForGivenComputeStage(ComputeStage.PRE,
+                CHAIN_TASK_ID, multipleCauses)).isTrue();
+        Assertions.assertThat(computeExitCauseService.getExitCausesAndPruneForGivenComputeStage(ComputeStage.PRE, CHAIN_TASK_ID))
+                .containsExactlyElementsOf(multipleCauses);
+
+        // After pruning, should return default unknown issue
+        Assertions.assertThat(computeExitCauseService.getExitCausesAndPruneForGivenComputeStage(ComputeStage.PRE,
```

Comment:
Same suggestion, it is easier to converge on a single assertion mechanism
I suggest to use `assertThat().isEqualTo()` and avoid `assertEquals`
And those can become static imports for a better readability in the code